### PR TITLE
improvement: activate fallback language only if getFallbackValues is set to true

### PR DESCRIPTION
**Pull request by bgpack**
## Changes in this pull request  
When` \Pimcore\Model\DataObject\Localizedfield::getGetFallbackValues()` is set to true the method `Translator->checkForEmptyTranslation()` should not use the fallback values for shared translations.

I added this check to force the `checkForEmptyTranslation` to return the translations id.

### CODE REVIEW

**Comment by martineiber** [comment]
Hey @bgpack 

Thank you for your contribution!

- Improvements should be merged into 11.x and not into 10.5.
- Can you please describe the Issue for this change. 
- It is not such a good idea to use the Localizedfield in the Translator, since both have nothing to do with each other.
- There should be a global setting for the fallback language.

**Comment by bgpack** [comment]
We use shared translations in object bricks selects. There we declare placeholder strings that will be translated by pimcore when we use the object bricks in a localized context. So far so good...
Further more we use a Data Quality mechanism similar to the one in your demo project. In the Data Quality we check for all used languages if the shared translation is present and if not we let the Data Quality fail.
The Data Quality code needs to compute the translations for the used placeholders to tell if all the shared translations are present... Now if pimcore ignores the framework state `\Pimcore\Model\DataObject\Localizedfield::getGetFallbackValues()` and returns the fallback values even if we declared that the `getFallbackValue == false` we will not be able the do the Data Quality.
Our Data Quality code sets `\Pimcore\Model\DataObject\Localizedfield::setGetFallbackValues(false)` in it's `compute()` code.
With this patch the Data Quality code is able to detect if shared translations are implemented or if they are still missing.

We are currently running pimcore 10.5 that's why we put the fix in 10.5.
Pimcore 6.9 did respect the framework state `\Pimcore\Model\DataObject\Localizedfield::getGetFallbackValues()`.
Would you please reconsider to put this fix to 10.5?

Regards,
Victor

**Comment by martineiber** [comment]
Your request is a project-specific request and not a pimcore-specific change request. Since this is an Edge case, we have to close your Pull Request and we cannot accept your change.

**Comment by bgpack** [comment]
@martineiber Hi Martin... In the documentation [Disable Fallback languages](https://pimcore.com/docs/pimcore/current/Development_Documentation/Objects/Object_Classes/Data_Types/Localized_Fields.html) you declare that pimcore will disable the Fallback language when we call `\Pimcore\Model\DataObject\Localizedfield::setGetFallbackValues(false);`.
pimcore is not behaving like this!

I - as a community member - would like to improve pimcore daily and don't understand why you declare my detailed explanation as project related, when the software doesn't behave according to the documentation.

The bug in pimcoreis that the command `\Pimcore\Model\DataObject\Localizedfield::setGetFallbackValues(false);` in the shared translations (messages) does not work and my code contribution fixes this situation.

**Comment by fashxp** [comment]
@bgpack we can understand your use case and your intention. 

Still, the docs you are referring to are related to localized fields and not to shared translations. Also the static function is in localized fields and it just feels wrong to use that flag for shared translations too. 

Besides that, as far as I can see, your suggested solution would not work in all the cases and cause some side effects and strange behavior. 
For example when loading a translation for a key with fallback languages active and later loading the translation for the same key without fallback languages will still return the fallback value (because it gets added to the catalog the first time). 
So also your data quality mechanism would eventually return false results. 

Bottom line: we just cannot merge that change. I guess it would require much more refactoring to cover all cases. 

Having that said: Might it be even better for your use case to check for missing translations directly in the translations table?
